Improve Travis CI setup and force CS rules

ste93cry opened the pull request:
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| License       | MIT

With this PR I would like to improve the build time in Travis CI and its setup. I've reworked the configuration so that we use the [stages](https://docs.travis-ci.com/user/build-stages/) feature and we cache the Composer directory to speed up things. I've also made Coveralls to run only if the build is successful as I think that knowing the code coverage when the build fails has no value

ste93cry committed:
Add and configure PHP CS Fixer

ste93cry committed:
Use the stages feature of Travis CI and improve the setup of the build

mavimo commented on .php_cs:
```diff
@@ -0,0 +1,25 @@
+        '@Symfony:risky' => true,
+        'random_api_migration' => true,
+        'ordered_imports' => true,
+        'yoda_style' => true,
```
IMHO yoda style make harder to read code, actually we should avoid involuntary assignation in a better way 😃

mavimo reviewed (changes requested):
We have a lot of (imho debatable) issues, maybe we should fix it or change the rules we have in PHPCS for header section.

ste93cry committed:
Fix the code style of all files

ste93cry commented:
Build is now green. I've moved the code coverage step to its own stage that runs only after all the tests pass, so we avoid waiting uselessly and fixed all CS issues according to the PHP-CS-Fixer configuration. Indeed some rules are debatable but I think that applying them now or later doesn't really change much as you will anyway probably do it at some point and regardless of the rules there will always be a bit commit full of changes. I've isolated these changes in its own commit by the way to ease the review.

mavimo committed:
Disable yoda style comparison
